### PR TITLE
AMP-42587 update segment plugin to use segment 4.1+

### PR DESCRIPTION
**Pull request by qingzhuozhen**
- Upgrade segment version
- Need to change the module name since the change introduced in [4.1.0](https://segment.com/docs/connections/sources/catalog/libraries/mobile/ios/changelog/#null)

### CODE REVIEW

**Review by justin-fiedler — APPROVED** [review]

**Review by justin-fiedler — CHANGES_REQUESTED** [review]
Just remembered we need to update the SegmentPlugin. Cartfile as well.

**Comment by qingzhuozhen** [comment]
Good catch @justin-fiedler let me update for Carthage framework as well, didn't install it earlier since we normally just use Cocoapod

File: SegmentPlugin/Cartfile [hunk 1]
```diff
@@ -1 +1 @@
-github "segmentio/analytics-ios" ~> 4.0.5
+github "segmentio/analytics-ios" ~> 4.1.0
```
Comment (justin-fiedler) [inline SegmentPlugin/Cartfile:1]: Same here. `~> 4.1`

**Review by justin-fiedler — CHANGES_REQUESTED** [review]

File: ItlySegmentPlugin.podspec [hunk 2]
```diff
@@ -19,7 +19,7 @@ Pod::Spec.new do |spec|
   spec.tvos.deployment_target = '13.4'
 
   spec.framework  = "Foundation"
-  spec.dependency "Analytics", "~> 4.0.0"
+  spec.dependency "Analytics", "~> 4.1.0"
```
[the remark this answers is not in the document]
Comment (qingzhuozhen) [reply ItlySegmentPlugin.podspec:22]: ok, will update that, hopefully they will honor the versioning and not bringing in breaking changes

**Review by justin-fiedler — APPROVED** [review]

**Pull request merged**